sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalUtils.scala

```diff
@@ -92,6 +92,36 @@ object IntervalUtils {
   }
 
   private val yearMonthPattern = "^([+|-])?(\\d+)-(\\d+)$".r
+  private val yearMonthStringPattern =
+    "(INTERVAL )?([+|-])*?(')?([+|-])?(\\d+)-(\\d+)(')?( YEAR TO MONTH)?".r
```

Comment:
`INTERVAL\s+`

Reply:
`([+|-])*`? The sign can only appear once, right?

Reply:
we also need `^` and `$`

Reply:
> `([+|-])*`? The sign can only appear once, right?


This is written to match as right as possible， as past pattern `-0-0`, the sign  will be match to prefix sign, but as current pattern, we don't need to do this now.

Reply:
DOne